src/main/java/rs/teslaris/core/assessment/service/impl/PublicationSeriesIndicatorServiceImpl.java

```diff
@@ -1,41 +1,550 @@
             publicationSeriesId,
             accessLevel).stream().map(
             EntityIndicatorConverter::toDTO).collect(Collectors.toList());
     }
+
+    @Override
+    public void computeFiveYearIFRank(List<Integer> classificationYears) {
+        int pageNumber = 0;
+        int chunkSize = 10;
+        boolean hasNextPage = true;
+
+        while (hasNextPage) {
+            List<JournalIndex> chunk =
+                journalIndexRepository.findAll(PageRequest.of(pageNumber, chunkSize)).getContent();
+
+            chunk.forEach((journalIndex) -> {
+                var currentJournal = journalService.findJournalById(journalIndex.getDatabaseId());
+                classificationYears.forEach(classificationYear -> {
+                    var currentJournalRankIndicators =
+                        publicationSeriesIndicatorRepository.findIndicatorsForPublicationSeriesAndCodeAndSourceAndYear(
+                            journalIndex.getDatabaseId(), "currentJIFRank", classificationYear,
+                            EntityIndicatorSource.WEB_OF_SCIENCE);
+
+                    if (currentJournalRankIndicators.isEmpty()) {
+                        return;
+                    }
+
+                    var distinctCategoryIdentifiers = currentJournalRankIndicators.stream()
+                        .map(PublicationSeriesIndicator::getCategoryIdentifier)
+                        .filter(Objects::nonNull)
+                        .filter(categoryIdentifier -> !categoryIdentifier.isEmpty())
+                        .distinct()
+                        .toList();
+
+                    distinctCategoryIdentifiers.forEach(categoryIdentifier -> {
+                        performIF5RankComputation(classificationYear, currentJournal,
+                            categoryIdentifier,
+                            currentJournalRankIndicators.getFirst().getEdition());
+                    });
+                });
+            });
+
+            pageNumber++;
+            hasNextPage = chunk.size() == chunkSize;
+        }
+    }
+
+    private void performIF5RankComputation(Integer classificationYear, Journal currentJournal,
+                                           String categoryIdentifier, String edition) {
+        var journalInSameCategoryIds =
+            publicationSeriesIndicatorRepository.findIndicatorsForCategoryAndYearAndSource(
+                categoryIdentifier, classificationYear, EntityIndicatorSource.WEB_OF_SCIENCE);
+        var allIF5Values =
+            publicationSeriesIndicatorRepository.findJournalIndicatorsForIdsAndCodeAndYearAndSource(
+                journalInSameCategoryIds, "fiveYearJIF", classificationYear,
+                EntityIndicatorSource.WEB_OF_SCIENCE);
+
+        var sortedIF5Values = allIF5Values.stream()
+            .filter((indicator) -> Objects.nonNull(indicator.getNumericValue()) &&
+                (indicator.getEdition().equals("SSCI") || indicator.getEdition().equals("SCIE")))
```

Comment:
This will be changed in June 2025 (probably ESCI and AHCI will be taken into account)